fitbenchmarking/cli/tests/test_checkpoint_handler.py

```diff
@@ -65,3 +70,340 @@ def test_create_files(self):
                     file = results / e
                     assert file.exists(), f'Failed to find "{e}"'
                     assert file.stat().st_size > 0, f'File is empty: "{e}"'
+
+
+class TestMergeDataSets(TestCase):
+    """
+    Simple testing for the checkpoint_handler merge entry point.
+    Uses the definition of identical defined in the docstring of
+    merge_data_sets
+    """
+
+    def setUp(self):
+        """
+        Generate 3 files to merge
+        """
+        ch_test_files = Path(__file__).parent / 'test_files'
+        self.A = str(ch_test_files/'A.json')
+        self.B = str(ch_test_files/'B.json')
+        self.C = str(ch_test_files/'C.json')
+        self.expected = ch_test_files / 'AB.json'
+        self._dir = TemporaryDirectory()  # pylint:disable=consider-using-with
+        self.dir = Path(self._dir.name)
+
+    def test_create_new_file(self):
+        """
+        Test that a new checkpoint file is generated.
+        """
+        output = self.dir / 'out.json'
+        merge_data_sets([self.A, self.B, self.C], str(output))
+        assert output.exists(), f'No new checkpoint file {output}'
+
+    def test_three_files(self):
+        """
+        Test that merging A, B, and C is the same as merging A and B, then
+        merging the result with C.
+        """
+        intermediate = self.dir / 'intermediate.json'
+        merge_data_sets([self.A, self.B], str(intermediate.absolute()))
+        twomerge = self.dir / 'two-merges.json'
+        merge_data_sets([str(intermediate), self.C], str(twomerge))
+        onemerge = self.dir / 'one-merge.json'
+        merge_data_sets([self.A, self.B, self.C], str(onemerge))
+
+        errors = []
+        for i, (l1, l2) in enumerate(zip(onemerge.read_text().splitlines(),
+                                         twomerge.read_text().splitlines())):
+            if l1 != l2:
+                errors.append((i, l1, l2))
```

Comment:
**[Optional refactoring suggestion]** This part can be extracted into a method and reused. Might save a few lines of code.

Reply:
If you don't feel strongly on this I'll leave it. I tend to favour tests that have as few functions as possible to keep them simple.

It is a good suggestion though and I'm happy to go with it if you think it should be fixed.

Reply:
No I think it is okay in that case to leave it as it is in that case 👍🏼